test/exec/test_variant_sender.cpp

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) Maikel Nadolski
+ * Copyright (c) 2023 NVIDIA Corporation
+ *
+ * Licensed under the Apache License Version 2.0 with LLVM Exceptions
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *   https://llvm.org/LICENSE.txt
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <exec/variant_sender.hpp>
+
+#include <catch2/catch.hpp>
+
+using namespace stdexec;
+using namespace exec;
+
+template <class... Ts>
+struct overloaded : Ts... {
+  using Ts::operator()...;
+};
+template <class... Ts>
+overloaded(Ts...) -> overloaded<Ts...>;
+
+TEST_CASE("variant sender", "[types][variant_sender]") {
+  using just_int_t = decltype(just(0));
+  using just_t = decltype(just());
+  variant_sender<just_t, just_int_t> variant = just();
+  int index = -1;
+  STATIC_REQUIRE(sender<variant_sender<just_t, just_int_t>>);
+  sync_wait(variant | then([&index](auto... xs) { index = sizeof...(xs); }));
+  CHECK(index == 0);
+
+  variant = just(42);
+  auto [value] = *sync_wait(
```

Comment:
Prefer using `opt.value()` in tests rather than `*opt`. The former throws, failing noisily when the optional is empty. `*opt` is just undefined behavior.